frontend/src/metabase/metadata/pages/DataModel/components/TablePicker/utils.ts

```diff
@@ -93,17 +107,27 @@ export function useTableLoader(path: TreePath) {
       if (databaseId === undefined || schemaName === undefined) {
         return [];
       }
-      const res = await fetchTables(
-        {
-          id: databaseId,
-          schema: schemaName,
-          include_hidden: true,
-          include_editable_data_model: true,
-        },
-        true,
-      );
+
+      const newArgs = {
+        id: databaseId,
+        schema: schemaName,
+        include_hidden: true,
+        include_editable_data_model: true,
+      };
+
+      if (
+        _.isEqual(tablesRef.current.originalArgs, newArgs) &&
+        tablesRef.current.isError
```

Comment:
Nit: put the least likely and cheapest condition first, it's easier to read and more performant (although it probably does not matter that much in this case).

```suggestion
        tablesRef.current.isError &&
        _.isEqual(tablesRef.current.originalArgs, newArgs)
        
```

Reply:
e21a7fa6097f633f30dbcb5bd4f54c5a4cdeb30b